Networking/Networking/Model/Product/ProductCategoryFromBatchCreation.swift

```diff
@@ -0,0 +1,69 @@
+import Foundation
+
+/// Represents a ProductCategory entity, used during categories batch creation.
+///
+struct ProductCategoryFromBatchCreation: Decodable {
+    let categoryID: Int64
+    let siteID: Int64
+    let parentID: Int64
+    let error: CreateError?
+    let name, slug: String?
```

Comment:
❓ Should `name` be non-optional, though? Is there any case that it can be optional that I'm missing?

Reply:
Good question. 
- `name` and `slug` will be nil when we receive an error. 
- In that case `error: CreateError` will be non-nil and `name` and `slug` will be nil

Sample response
```
[
      {
        "id": 0,
        "error": {
          "code": "term_exists",
          "message": "A term with the name provided already exists with this parent.",
          "data": {
            "status": 400,
            "resource_id": 1373
          }
        }
      },
      {
        "id": 1377,
        "name": "Other",
        "slug": "other",
        // ...
        }
      }
]
```

Reply:
If we receive an error, should the mapper fail? I don't understand why we should map a model with an error inside.

Reply:
Let me try to explain my understanding. 

- We send five categories in the create request.
- 2 of them already exist in remote so we receive error for those two.
- 3 categories are created so we receive the new categories.
- The response will totally have 5 JSON elements.
- When we parse the response, we should parse the error items along with the 3 new categories.
- Later, we ignore the error items https://github.com/woocommerce/woocommerce-ios/pull/10846/files#diff-614fee19aa6e48eb4075b6cf58ca0843f619d06793b6feef495754f1ad0c1e8dR40

Reply:
I will be merging this PR now to unblock other child PRs. I will address any comments in a future PR if needed. Thank you!

Reply:
Thanks for the explanation. I'm curious why we have to send all categories for bulk creation. Isn't it possible to send only the ones that don't already exist?

Reply:
> I'm curious why we have to send all categories for bulk creation. Isn't it possible to send only the ones that don't already exist?

Good question. 

We will be sending only the local/new categories in the creation request. However, there is a possibility that a category with the same name and parent is created by another user, which will throw a `term_exists` error for that item. 

Explained here in this internal link - p1696499801271349/1696499095.943969-slack-C03L1NF1EA3

Reply:
Gotcha, thanks!